### DISCUSSION TITLE
Use descriptive names for mouse buttons

### DISCUSSION THREAD

**Discussion by timothyqiu** [godotengine/godot-proposals#5539] [GUI]
`BUTTON_LEFT` and `BUTTON_RIGHT` are actually referring to the primary button and secondary button. i.e. If you set your primary button to right button in OS settings, `BUTTON_LEFT` will be reported for the right mouse button.

I think we should either document this, or rename them accordingly.

**Comment by Mickeon** [comment]
If they're renamed to MOUSE_PRIMARY and MOUSE_SECONDARY, it would be strange to keep the other mouse buttons the same. At that point the mouse buttons may as well be less verbose and ordered, like it is convention in some games _(thinking of Source games' `M1` to `M5`)_. Though that would not fit Godot's priority of being explicit first. I feel it should be documented. I've not heard the inverted mouse buttons to be all that commonly used.
